File: src/2784.check-if-array-is-good.2892/solution.ts
function isGood (nums: number[]): boolean {
    const N = nums.length - 1;
    const counts = new Array<number>(N).fill(0);
    for (const num of nums) {
        if (num > N || num < 1) {
            return false;
        }
        const index = num - 1;
        if (num !== N && counts[index] === 1) {
            return false;
        }
        if (num === N && counts[index] === 2) {
            return false;
        }
        counts[index]++;
    }
    return true;
}

File: src/2784.check-if-array-is-good.2892/solution.java
class Solution {
    public boolean isGood(int[] nums) {
        int N = nums.length-1;
        boolean[] exists = new boolean[N];
        int cnt = 0;
        for(int n : nums){
            if(n>N){
                return false;
            }
            if(n == N){
                cnt++;
                continue;
            }
            if(exists[n]){
                return false;
            }
            exists[n] = true;
        }
        
        return cnt == 2;
    }
}